src/shapes/SurfaceShape.js

```diff
@@ -425,17 +437,61 @@ define([
 
         // Internal. Intentionally not documented.
         SurfaceShape.prototype.intersectsFrustum = function (dc) {
-            if (this._extent) {
+            if (this.currentData && this.currentData.extent) {
                 if (dc.pickingMode) {
-                    return this._extent.intersectsFrustum(dc.pickFrustum);
+                    return this.currentData.extent.intersectsFrustum(dc.pickFrustum);
                 } else {
-                    return this._extent.intersectsFrustum(dc.navigatorState.frustumInModelCoordinates);
+                    return this.currentData.extent.intersectsFrustum(dc.navigatorState.frustumInModelCoordinates);
                 }
             } else {
                 return true;
             }
         };
 
+        /**
+         * Indicates whether a specified shape data object is current. Subclasses may override this method to add
+         * criteria indicating whether the shape data object is current, but must also call this method on this base
+         * class. Applications do not call this method.
+         * @param {DrawContext} dc The current draw context.
+         * @param {Object} shapeData The object to validate.
+         * @returns {Boolean} true if the object is current, otherwise false.
+         * @protected
+         */
+        SurfaceShape.prototype.isShapeDataCurrent = function (dc, shapeData) {
+            return shapeData.verticalExaggeration === dc.verticalExaggeration
+                && shapeData.expiryTime > Date.now();
+        };
+
+        /**
+         * Creates a new shape data object for the current globe state. Subclasses may override this method to
+         * modify the shape data object that this method creates, but must also call this method on this base class.
+         * Applications do not call this method.
+         * @returns {Object} The shape data object.
+         * @protected
+         */
+        SurfaceShape.prototype.createShapeDataObject = function () {
+            return {};
+        };
+
+        // Intentionally not documented.
+        SurfaceShape.prototype.resetExpiration = function (shapeData) {
+            // The random addition in the line below prevents all shapes from regenerating during the same frame.
+            shapeData.expiryTime = Date.now() + this.expirationInterval + 1e3 * Math.random();
+        };
+
+        // Internal. Intentionally not documented.
+        SurfaceShape.prototype.establishCurrentData = function (dc) {
+            this.currentData = this.shapeDataCache.entryForKey(dc.globeStateKey);
+            if (!this.currentData) {
+                this.currentData = this.createShapeDataObject();
+                this.resetExpiration(this.currentData);
+                this.currentData.verticalExaggeration = dc.verticalExaggeration;
```

Comment:
Is the VE assignment here still needed?

Reply:
@pdavidc I suppose we can remove it which will just cause the currentData to immediately expire and get reset on the next render.

Reply:
The extent will be null anyway and so will cause computeExtent, etc. to fire.